### DISCUSSION TITLE
Custom decoding of an integer with pgx/v5

### DISCUSSION THREAD

**Discussion by sebastianwestberg** [jackc/pgx#1560] [Q&A] [answered]
Hi!

I've tried to implement multiple interfaces in order to create a custom decoder for one of my integer fields, without any luck.
The goal is to convert the integer from postgres, into a formatted string:

```
type IntAsString string

func (ias *IntAsString) DecodeBinary(ci *pgtype.ConnInfo, src []byte) error {
	panic("test")
}
```

I can't get the panic to trigger.
Same with the following attempt:
```
func (ias *IntAsString) Scan(value interface{}) error {
	panic("test")
}
```

Can I please get some guidance? Thanks in advance!

**Comment by jackc** [comment, marked as answer]
I think implementing `pgtype.Int64Scanner` and `pgtype.Int64Valuer` will do what you want.

**Reply by sebastianwestberg** [reply]
Thank you so much jackc, that's just what I needed. Leaving my code here in case it helps someone else.

```go
type IntAsFormattedString string

func (ias *IntAsFormattedString) ScanInt64(n pgtype.Int8) error {
	if !n.Valid {
		*ias = "00:00"
		return nil
	}

	// Convert the integer to a string in the format "mm:ss"
	minutes := n.Int64 / 60
	seconds := n.Int64 % 60
	*ias = IntAsFormattedString(fmt.Sprintf("%02d:%02d", minutes, seconds))

	return nil
}
```
